### ISSUE TITLE
[FEATURE] Be able to use environment variables for provider configuration

### ISSUE THREAD

**Issue by LvffY** [kota65535/terraform-provider-maven#8]
Hi

First of all : thanks for the provider, it saved me quite a time in my context :).

I'm quite new to terraform, but I think it could be useful to be able to define some custom environment variables to configure the provider. In our case, it's mostly useful when using some private maven repository.

For example, we can look at the documentation of the [databricks provider](https://registry.terraform.io/providers/databricks/databricks/0.2.3/docs#environment-variables) where each provider configuration is associated to an environment variable prefixed by **DATABRICKS_**.

Of course, as a workaround, we can do the following : 

```hcl
variable "maven_url" {
  type = string
  description = "repository url"
} 
variable "maven_username" {
  type = string
  description = "repository username to access"
} 
variable "maven_password" {
  type = string
  description = "repository password to access"
} 
provider "maven" {
  repository_url = var.maven_url
  username = var.maven_username
  password = var.maven_password
} 
```

and define respectively **TF_VAR_maven_url**, **TF_VAR_maven_username** and **TF_VAR_maven_password** but this is not really explicit and forces us to define some "useless" variables explicitly.

In this case, I think it could be quite useful to prefix these variables by **MAVEN_**.

**Comment by kota65535** [comment]
Sorry I didn't realize this issue for a long time. The feature is quite simple so I will implement this weekend.

**Comment by kota65535** [comment]
@LvffY I've released v0.2.1, please try it out.
Documents also updated: https://registry.terraform.io/providers/kota65535/maven/latest/docs#environment-variables
